private/buf/bufctl/controller.go

```diff
@@ -393,20 +394,49 @@ func (c *controller) GetTargetImageWithConfigs(
 			// We did not find a buf.yaml in our current directory, and there was no config override.
 			// Use the defaults.
 		} else {
-			switch fileVersion := bufYAMLFile.FileVersion(); fileVersion {
-			case bufconfig.FileVersionV1Beta1, bufconfig.FileVersionV1:
-				moduleConfigs := bufYAMLFile.ModuleConfigs()
-				if len(moduleConfigs) != 1 {
-					return nil, fmt.Errorf("expected 1 ModuleConfig for FileVersion %v, got %d", len(moduleConfigs), fileVersion)
+			// We take the top-level lint and breaking config from buf.yaml.
+			//
+			// In the case of v1 buf.yaml files, there is only ever one top-level config, so it
+			// should be populated and we use that.
+			// In the case of v2 buf.yaml files, this may be empty, since there may not be a
+			// top-level config. If that happens, we print a warning to the user and let them
+			// know that we are using v2 defaults.
+			// In the case where there is a top-level config and individual module configs, we
+			// print out a message to the user to let them know we are using the top-level config.
+			var configInfo string
+			// If configuration is read from an override, we determine if it was a raw config or
+			// a path to a valid override file.
+			if override := functionOptions.configOverride; override != "" {
+				switch filepath.Ext(override) {
+				case ".json", ".yaml", ".yml":
+					configInfo = functionOptions.configOverride
+				default:
+					configInfo = "from --config"
 				}
-				lintConfig = moduleConfigs[0].LintConfig()
-				breakingConfig = moduleConfigs[0].BreakingConfig()
-			case bufconfig.FileVersionV2:
-				// Do nothing. Use the default LintConfig and BreakingConfig. With
-				// the new buf.yamls with multiple modules, we don't know what lint or
-				// breaking config to apply.
-			default:
-				return nil, syserror.Newf("unknown FileVersion: %v", fileVersion)
+			} else {
+				// Otherwise we use the file path
+				configInfo = bufYAMLFile.ObjectData().Name()
+			}
+			if topLevelModuleConfig := bufYAMLFile.TopLevelModuleConfig(); topLevelModuleConfig != nil {
+				if bufYAMLFile.FileVersion() == bufconfig.FileVersionV2 && len(bufYAMLFile.ModuleConfigs()) > 1 {
+					c.logger.Sugar().Warnf(
```

Comment:
I don't think we need to print either of these warnings, I just think we take the position that doing lint or breaking change detection for an image means we treat it like any other module - if it doesn't have a configuration, we use the top-level configuration, which may be empty and then we just go to default.